pkg/csender/hub.go

```diff
@@ -0,0 +1,119 @@
+package csender
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+
+	"github.com/pkg/errors"
+)
+
+func (cs *Csender) httpClient() *http.Client {
+	tr := *(http.DefaultTransport.(*http.Transport))
+
+	if runtime.GOOS != "windows" && runtime.GOOS != "darwin" {
+		// inject ca root certs, cause they can be missing on some systems
+		rootCertsPath := "/etc/cagent/cacert.pem"
+		if _, err := os.Stat(rootCertsPath); err == nil {
```

Comment:
There is pretty similar code in cagent.New() function. We can move it to some common package like 'common'

Reply:
agree, done